application/src/main/java/core/application/filter/CustomLogoutFilter.java

```diff
@@ -0,0 +1,90 @@
+package core.application.filter;
+
+import core.application.security.TokenService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.web.filter.GenericFilterBean;
+
+import java.io.IOException;
+
+/**
+ * 사용자 로그아웃 요청을 처리하는 커스텀 필터
+ * Spring의 GenericFilterBean을 확장하여 로그아웃 시 Refresh 토큰 처리
+ */
+public class CustomLogoutFilter extends GenericFilterBean {
+    private final TokenService tokenService;
+
+    /**
+     * 생성자.
+     *
+     * @param tokenService 토큰 관리 서비스
+     */
+    public CustomLogoutFilter(TokenService tokenService) {
+        this.tokenService = tokenService;
+    }
+
+    /**
+     * 필터 체인에서 요청 필터링
+     *
+     * @param request HTTP 요청
+     * @param response HTTP 응답
+     * @param chain 필터 체인
+     * @throws IOException 입출력 오류 발생 시
+     * @throws ServletException 서블릿 관련 오류 발생 시
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        doFilter((HttpServletRequest) request, (HttpServletResponse) response, chain);
+    }
+
+    /**
+     * HTTP 요청을 필터링하고 로그아웃 로직 처리
+     *
+     * @param request HTTP 요청
+     * @param response HTTP 응답
+     * @param filterChain 필터 체인
+     * @throws IOException 입출력 오류 발생 시
+     * @throws ServletException 서블릿 관련 오류 발생 시
+     */
+    private void doFilter(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+
+        // 요청 URI 및 메소드 확인
+        String requestUri = request.getRequestURI();
+        if (!requestUri.matches("^/users/signout$")) {
+
+            // 로그아웃 경로가 아닌 경우 필터 체인 진행
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        String requestMethod = request.getMethod();
+        if (!requestMethod.equals("DELETE")) {
+
+            // DELETE 메소드가 아닌 경우 필터 체인 진행
+            filterChain.doFilter(request, response);
+            return;
+        }
```

Comment:
요청 API 가 `/users/singout` 이고 `HTTP DELETE` 인 경우만 걸러내는 게 맞나요?

그럼 아래처럼 바꿀 수 있을 것 같아요.

```suggestion
        // 요청 URI & HTTP 메서드 확인
        String requestUri = request.getRequestURI();
        String requestMethod = request.getMethod();
        if (!requestUri.matches("^/users/signout$") || !requestMethod.equals("DELETE")) {
            filterChain.doFilter(request, response);
            return;
        }
```

Reply:
네 이렇게 수정하는게 중복되는 코드가 사라지니 더 나을 것 같네요. 수정해서 반영하겠습니다.

Reply:
8178de5113997a7a62e24672de9604dc59873262

Reply:
4f966a5d6652ef0ac441cdb4872178c395f76e57